contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -832,20 +832,41 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
    * @notice median from the most recent report
    */
   function latestAnswer() public view virtual override returns (int256) {
-    return s_transmissions[s_hotVars.latestAggregatorRoundId].answer;
+    return s_transmissions[uint32(latestRound())].answer;
   }
 
   /**
    * @notice timestamp of block in which last report was transmitted
    */
   function latestTimestamp() public view virtual override returns (uint256) {
-    return s_transmissions[s_hotVars.latestAggregatorRoundId].recordedTimestamp;
+    return s_transmissions[uint32(latestRound())].recordedTimestamp;
   }
 
   /**
-   * @notice Aggregator round (NOT OCR round) in which last report was transmitted
+   * @notice Aggregator round (NOT OCR round) in which last valid report was transmitted
    */
   function latestRound() public view virtual override returns (uint256) {
```

Comment:
Since this public returns uint256, but the underlying round id is uint32, what I propose is creating an internal functions _getLatestRound() which returns uint32. 

latestRound() implementation will simply be:
return _getLatestRound().

This is also more gas efficient, because we won't be calling public functions internally. It will also prevent weird casting throughout the code.

Reply:
Good idea, I created the `_getLatestRound()` function and removed all the unnecessary casts.